pylint/pyreverse/writer.py

```diff
@@ -68,21 +71,42 @@ def write_packages(self, diagram: PackageDiagram) -> None:
                 type_=NodeType.PACKAGE,
                 properties=self.get_package_properties(module),
             )
+
+            module_info[module.fig_id] = {
+                "imports": 0,
+                "imported": 0,
+            }
+
         # package dependencies
         for rel in diagram.get_relationships("depends"):
+            from_id = rel.from_object.fig_id
+            to_id = rel.to_object.fig_id
+
             self.printer.emit_edge(
-                rel.from_object.fig_id,
-                rel.to_object.fig_id,
+                from_id,
+                to_id,
                 type_=EdgeType.USES,
             )
 
+            module_info[from_id]["imports"] += 1
+            module_info[to_id]["imported"] += 1
+
         for rel in diagram.get_relationships("type_depends"):
+            from_id = rel.from_object.fig_id
+            to_id = rel.to_object.fig_id
+
             self.printer.emit_edge(
-                rel.from_object.fig_id,
-                rel.to_object.fig_id,
+                from_id,
+                to_id,
                 type_=EdgeType.TYPE_DEPENDENCY,
             )
 
+            module_info[from_id]["imports"] += 1
+            module_info[to_id]["imported"] += 1
+
+        print(f"Modules: {len(module_info)}")
+        print(f'Imports: {sum(mod["imports"] for mod in module_info.values())}')
```

Comment:
Not sure if it should be in non verbose mode but I'll let Andreas decide.

```suggestion
        if verbose: # TODO add verbose in this function parameters if we do that
            print(f"Modules: {len(module_info)}")
            print(f'Imports: {sum(mod["imports"] for mod in module_info.values())}')
```

Reply:
My thought was to print that by default. After all, it isn't debugging information, it's just information. There are questions about module relations that Pyreverse is well-suited to answer since it's looking at module relations anyway. So I figure, why not print it?

There are lots of other stats that could be added. Which module is the most / least imported? Which module has the most / fewest imports? How many transitive dependencies are there? I've found that some design and organization improvements can reveal themselves through investigating anomalous import stats.

Reply:
I guess clever warning about module stats should go in pylint or another standalone tool (there's a bunch of clever high level design check that are only in the litterature on arxiv or design books) . I see pyreverse more as a graphical display of uml diagram

Reply:
As for the output I would print it by default, I feel it gives a nice concise feedback how much was analysed. 
What do you think of printing it like "Analysed X modules with a total of Y imports"?
Alternatively at least:
```
Analysed modules: X
Total imports: Y
```


As for the other stats, I agree with Pierre. This can be very useful information, but I don't think it belongs in `pyreverse`. If at all in Pylint and not a dedicated **design**. checker, it would be more of a use case for a [report](https://pylint.readthedocs.io/en/stable/user_guide/usage/output.html#reports-section).

Reply:
I like "Analysed X modules with a total of Y imports" in one line 👍